packages-native/actions/src/camera/TakePictureAdvanced.ts

```diff
@@ -0,0 +1,225 @@
+ * @param {Big} maximumHeight - The picture will be scaled to this maximum pixel height, while maintaing the aspect ratio.
+ * @param {MxObject} pictureData - Additional info about the picture will be stored in this object. Create it before calling this action.
+ * @returns {Promise.<mendix.lib.MxObject>}
+ */
+export async function TakePictureAdvanced(
+    picture?: mendix.lib.MxObject,
+    pictureSource?: PictureSource,
+    pictureQuality?: PictureQuality,
+    maximumWidth?: BigJs.Big,
+    maximumHeight?: BigJs.Big
+): Promise<boolean> {
+    // BEGIN USER CODE
+    // Documentation https://github.com/react-native-community/react-native-image-picker/blob/master/docs/Reference.md
+
+    return new Promise((resolve, reject) => {
+        if (!picture) {
+            return reject(new Error("Input parameter 'Picture' is required"));
+        }
+        if (!picture.inheritsFrom("System.FileDocument")) {
+            const entity = picture.getEntity();
+            return reject(new Error(`Entity ${entity} does not inherit from 'System.FileDocument'`));
+        }
+        if (pictureQuality === "custom" && !maximumHeight && !maximumWidth) {
+            return reject(new Error("Picture quality is set to 'Custom', but no maximum width or height was provided"));
+        }
+        createMxObject("NativeMobileResources.TakePictureData").then((resultObject: any) => {
+            takePicture()
+                .then((response: any) => {
+                    if (!response || !response.uri) {
+                        return resolve(resultObject);
+                    }
+                    storeFile(picture, response.uri).then(pictureTaken => {
+                        resultObject.set("PictureTaken", pictureTaken);
+                        resultObject.set("URI", response.uri);
+                        resultObject.set("IsVertical", response.isVertical);
+                        resultObject.set("Width", response.width);
+                        resultObject.set("Height", response.height);
+                        resultObject.set("FileName", response.fileName);
+                        resultObject.set("FileSize", response.fileSize);
+                        resultObject.set("FileType", response.type);
+                        resolve(resultObject);
+                    });
+                })
+                .catch(error => {
+                    if (error === "canceled") {
+                        resolve(resultObject);
+                    } else {
+                        throw new Error(error);
+                    }
+                });
+        });
+    });
+
+    function takePicture(): Promise<string | undefined> {
+        return new Promise((resolve, reject) => {
+            const options = getOptions();
+            const method = getPictureMethod();
+
+            method(options, response => {
+                if (response.didCancel) {
+                    return resolve();
```

Comment:
I think as you are defining a new promise, the same needs to be resolve and not return another promise. If you will not return the promise you can you Promise.resolve instead.